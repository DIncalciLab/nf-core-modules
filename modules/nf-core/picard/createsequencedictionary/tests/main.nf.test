nextflow_process {

    name "Test Process PICARD_CREATESEQUENCEDICTIONARY"
    script "../main.nf"
    process "PICARD_CREATESEQUENCEDICTIONARY"

    tag "modules"
    tag "modules_nfcore"
    tag "picard"
    tag "picard/createsequencedictionary"

    test("sarscov2 - fasta") {

        when {
            process {
                """
                input[0] = [
                    [ id:'test' ], // meta map
                    file(params.modules_testdata_base_path + 'genomics/sarscov2/genome/genome.fasta', checkIfExists: true)]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out.reference_dict).match("reference_dict") },
                { assert snapshot(process.out.versions).match("versions") }

            )
        }

    }

    test("sarscov2 - fasta - stub") {

        options "-stub"

        when {
            process {
                """
                input[0] = [
                    [ id:'test' ], // meta map
                    file(params.modules_testdata_base_path + 'genomics/sarscov2/genome/genome.fasta', checkIfExists: true)]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out.reference_dict).match("stub_reference_dict") },
                { assert snapshot(process.out.versions).match("stub_versions") }
            )
        }

    }

}
