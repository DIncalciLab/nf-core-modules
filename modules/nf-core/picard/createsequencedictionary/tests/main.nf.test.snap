{
    "sarscov2 - fasta - stub": {
        "content": [
            "test.dict",
            [
                "versions.yml:md5,fdb110d8c303ffa0615bb200f9a2c999"
            ]
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.04.4"
        },
        "timestamp": "2024-08-16T14:26:10.223708977"
    },
    "sarscov2 - fasta": {
        "content": [
            "test.dict",
            [
                "versions.yml:md5,fdb110d8c303ffa0615bb200f9a2c999"
            ]
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.04.4"
        },
        "timestamp": "2024-08-16T14:31:34.330709472"
    }
}